[BUG] Task `:distribution:bwc:minor:buildBwcLinuxArm64Tar` fails on Linux ARM

### Describe the bug

```
> Task :distribution:bwc:minor:buildBwcLinuxArm64Tar FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':distribution:bwc:minor:buildBwcLinuxArm64Tar'.
> Building 2.19.0 didn't generate expected file /home/ubuntu/OpenSearch/distribution/bwc/minor/build/bwc/checkout-2.x/distribution/archives/linux-arm64-tar/build/distributions/opensearch-min-2.19.0-SNAPSHOT-linux-arm64-x64.tar.gz
```

### Related component

Build

### To Reproduce

Execute `./gradlew :distribution:bwc:minor:buildBwcLinuxArm64Tar` on a Graviton AWS EC2 instance with Ubuntu installed.

It expects the file `opensearch-min-2.19.0-SNAPSHOT-linux-arm64-x64.tar.gz` but the actual file that is created is `opensearch-min-2.19.0-SNAPSHOT-linux-arm64.tar.gz` (there's an extra "-x64" in the expected file).

I believe the problem is here:

https://github.com/opensearch-project/OpenSearch/blob/d158ec6d431615b16192c34490b46752f3eb3e0f/buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionBwcSetupPlugin.java#L184

If I remove that trailing `"-x64"` then it works, though I'm sure that change would break the build for the non-ARM case.

### Expected behavior

task succeeds

### Additional Details

`m8g.4xlarge` EC2 instance running Ubuntu 24.04

--- a/buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionBwcSetupPlugin.java
+++ b/buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionBwcSetupPlugin.java
@@ -181,15 +181,19 @@ public class InternalDistributionBwcSetupPlugin implements Plugin<Project> {
                 if (name.contains("zip") || name.contains("tar")) {
                     int index = name.lastIndexOf('-');
                     String baseName = name.substring(0, index);
-                    classifier = "-" + baseName + "-x64";
+                    classifier = "-" + baseName;
+                    // The x64 variants do not have the architecture built into the task name, so it needs to be appended
+                    if (name.equals("darwin-tar") || name.equals("linux-tar") || name.equals("windows-zip")) {
+                        classifier += "-x64";
+                    }
                     extension = name.substring(index + 1);
                     if (extension.equals("tar")) {
                         extension += ".gz";
                     }
                 } else if (name.contains("deb")) {
-                    classifier = "-amd64";
+                    classifier = "_amd64";
                 } else if (name.contains("rpm")) {
-                    classifier = "-x64";
+                    classifier = ".x86_64";
                 }
             } else {
                 extension = name.substring(4);
@@ -256,9 +260,21 @@ public class InternalDistributionBwcSetupPlugin implements Plugin<Project> {
             this.name = name;
             this.projectPath = baseDir + "/" + name;
             if (version.onOrAfter("1.1.0")) {
+                // Deb uses underscores (I don't know why...):
+                // https://github.com/opensearch-project/OpenSearch/blob/f6d9a86f0e2e8241fd58b7e8b6cdeaf931b5108f/distribution/packages/build.gradle#L139
+                final String separator = name.equals("deb") ? "_" : "-";
                 this.distFile = new File(
                     checkoutDir,
-                    baseDir + "/" + name + "/build/distributions/opensearch-min-" + version + "-SNAPSHOT" + classifier + "." + extension
+                    baseDir
+                        + "/"
+                        + name
+                        + "/build/distributions/opensearch-min"
+                        + separator
+                        + version
+                        + "-SNAPSHOT"
+                        + classifier
+                        + "."
+                        + extension
                 );
             } else {
                 this.distFile = new File(

